src/js/modules/supervisors/Repository.ts

```diff
@@ -87,31 +90,68 @@ class Repository {
     return this.handles.size;
   }
 
+  createEmptyProcessBatchReplay(
+    handle: Handle,
+    requestItem: ProcessBatchRequestItem
+  ): ProcessBatchReplyItem {
+    return {
+      coprocessorId: BigInt(handle.coprocessor.globalId),
+      ntp: {
+        ...requestItem.ntp,
+        topic: `${requestItem.ntp.topic}`,
+      },
+      resultRecordBatch: [],
+    };
+  }
+
   transformMapResultToArray(
     requestItem: ProcessBatchRequestItem,
     handle: Handle,
     recordBatch: RecordBatch,
     resultRecordBatch: Map<string, RecordBatch>
   ): ProcessBatchReplyItem[] {
     const results: ProcessBatchReplyItem[] = [];
-    for (const [key, value] of resultRecordBatch) {
-      value.records = value.records.map((record) => {
-        record.length = calculateRecordLength(record);
-        record.valueLen = record.value.length;
-        return record;
-      });
-      value.header.sizeBytes = calculateRecordBatchSize(value.records);
-      value.header.term = recordBatch.header.term;
-      results.push({
-        coprocessorId: BigInt(handle.coprocessor.globalId),
-        ntp: {
-          ...requestItem.ntp,
-          topic: `${requestItem.ntp.topic}.$${key}$`,
-        },
-        resultRecordBatch: [value],
-      });
-      return results;
+    if (resultRecordBatch === undefined) {
+      /*
+       Coprocessor doesn't return anything, in this case, it responses to
```

Comment:
it responds with an empty process batch to signal that it shouldn't send retry sending this request again.*

Reply:
Yes, I'll apply the @senior7515 approach, if the coprocessor doesn't return anything, that coprocessor is going to remove and disable.